File: scripts/day1-part2-simple.py
import time

def solve_part2(rotations):
    """Simple, straightforward version"""
    position = 50
    count = 0
    
    for rotation in rotations:
        direction = rotation[0]
        distance = int(rotation[1:])
        
        if direction == 'L':
            # Count how many times we pass through 0 going left
            for _ in range(distance):
                position = (position - 1) % 100
                if position == 0:
                    count += 1
        else:
            # Count how many times we pass through 0 going right
            for _ in range(distance):
                position = (position + 1) % 100
                if position == 0:
                    count += 1
    
    return count

# Read input
with open("../inputs/day-1-input.txt", "r") as f:
    input_data = f.read().strip().split('\n')

# Benchmark
iterations = 1
start_time = time.perf_counter()

for _ in range(iterations):
    result = solve_part2(input_data)

end_time = time.perf_counter()
total_time = end_time - start_time
avg_time_us = (total_time / iterations) * 1_000_000

print(f"Python (Day 1 Part 2 - Simple)")
print(f"Answer: {result}")
print(f"Average time: {avg_time_us:.3f} microseconds ({iterations} iterations)")
print(f"Total time: {total_time * 1000:.2f} ms")

File: scripts/day1-part2-simple.js
const fs = require("fs");

// Simple, straightforward version
function solvePart2(rotations) {
  let position = 50;
  let count = 0;

  for (const rotation of rotations) {
    const direction = rotation[0];
    const distance = parseInt(rotation.substring(1));

    if (direction === "L") {
      // Count how many times we pass through 0 going left
      for (let i = 1; i <= distance; i++) {
        position = (position - 1) % 100;
        if (position < 0) position += 100;
        if (position === 0) count++;
      }
    } else {
      // Count how many times we pass through 0 going right
      for (let i = 1; i <= distance; i++) {
        position = (position + 1) % 100;
        if (position === 0) count++;
      }
    }
  }

  return count;
}

const input = fs
  .readFileSync("../inputs/day-1-input.txt", "utf-8")
  .trim()
  .split(/\r?\n/);

const iterations = 1;
const startTime = process.hrtime.bigint();

let result;
for (let i = 0; i < iterations; i++) {
  result = solvePart2(input);
}

const endTime = process.hrtime.bigint();
const totalNanos = Number(endTime - startTime);
const avgMicros = (totalNanos / iterations / 1000).toFixed(3);

console.log(`JavaScript (Day 1 Part 2 - Simple)`);
console.log(`Answer: ${result}`);
console.log(`Average time: ${avgMicros} microseconds (${iterations} iterations)`);
console.log(`Total time: ${(totalNanos / 1000000).toFixed(2)} ms`);
